python/cuml/cuml/internals/api_decorators.py

```diff
@@ -193,10 +193,6 @@ def wrapper(*args, **kwargs):
                         if self_val is None:
                             assert input_val is not None
                             out_type = iu.determine_array_type(input_val)
-                        elif input_val is None or input_val is inspect._empty:
-                            out_type = self_val.output_type
-                            if out_type == "input":
-                                out_type = self_val._input_type
```

Comment:
Can you comment on what motivated the removal here? There is a lot of indirection in the entire reflection system, so it's not immediately clear to me how we changed the logic here and what prompted this specific change in combination with the changes to the base class.

Reply:
AFAICT that could never be hit (maybe it used to be possible but has since changed).